Python/codecs.c

```diff
@@ -147,7 +147,12 @@ PyObject *_PyCodec_Lookup(const char *encoding)
     if (v == NULL) {
         return NULL;
     }
-    PyUnicode_InternInPlace(&v);
+
+    /* Intern the string.
+       A successful lookup will insert `v` in the codecs search cache,
+       effectively immortalizing it. We don't want to immortalize the key
+       when lookup fails. */
+    _PyUnicode_InternMortal(interp, &v);
```

Comment:
`PyDict_SetItem` is used to insert `v` instead of `PyDict_SetItemString`, so I'm not sure where `v` is immortalized.